### ISSUE TITLE
A small question about collapse isoforms script

### ISSUE THREAD

**Issue by WayneForever** [Magdoll/cDNA_Cupcake#44]
Hi,

Firstly, thanks for designing this great software. I'm now working on one project in which the script 'collapse_isoforms_by_sam.py' is used for reducing redundancy of the GMAP result. 

During the first time of running the script, I used the default parameters of minimal coverage and minimal identity. The final result shows that I got 288,773 collapse isoforms and 13,354 genes. 

I personally thought that the number of collapse isoforms is small due to the high filter rate (42.75%). Therefore I changed the coverage and identity value to 0.85 and 0.9 respectively. The final result shows 476,088 collapse isoforms with the filter rate of 16.86%.  However, I found that this time I got only 1,573 genes based on the file collapsed.group.txt. 

In other words, I got more isoforms but fewer genes after changing the two parameters. 

I'm still a beginner at bioinformatics and it's a little bit hard for me to find the reason from the script. Therefore, I'm writing this issue to figure out the reason. My personal guess is that after changing the parameters, I got more isoforms and on the other hand, the way of dividing gene groups changed as well. 

I'm looking forward to waiting for answers and thank you very much.

**Comment by Magdoll** [comment]
Hi @WayneForever ,

Yes you are correct that the estimation of genes (technically, just "loci") changes.

This is because the collapse script doesn't actually know where genes are since no annotation is used. It simply creates "loci", where the definition is "a collection of overlapping transcripts on the same strand, even if they only overlap by 1 bp"

So, with more transcripts, it's possible now there are more overlaps and hence fewer "genes".

--Liz

**Comment by WayneForever** [comment]
Thanks for your reply, Liz.

**Comment by WayneForever** [comment]
Hi, @Magdoll 

Sorry for bothering you again. There is another thing that I don't understand. 

This project follows the Isoseq2 from Pacbio where 16 cells were sequenced. In total, I got 1,267,862 reads and 288,773 isoforms based on 'collapse_isoforms_by_sam.py' script. The resulting file 'all.collapsed.group.txt' shows that some loci (The gene I mentioned above) has even hundreds of isoforms such as PB.1.259 or PB.63.242. 

In most circumstances, this situation seldom happens because the number of isoforms in one gene shouldn't be that much. Can this situation happen if my input of the script is correct? 

My script runs as follows, and the file 'rm_fusion.mapped.sorted.sam' is the gmap.sam without fusion gene:
`python collapse_isoforms_by_sam.py --input  NGS_corrected.fasta -s rm_fusion.mapped.sorted.sam -o all`

Sorry again for my bothering and thank you very much.

**Comment by Magdoll** [comment]
Hi @WayneForever ,

So you had 288,773 HQ isoforms (after running Iso-Seq2 pipeline) and then ran collapsed?

I would look at the before (`rm_fusion.mapped.sorted.sam`) and after (`all.collapsed.gff`) on a browser (ex: IGV) and look if the high splicing diversity for those genes is really there. It's entirely possible, especially for neural genes in human, to have a lot of isoforms. One of the first Iso-Seq papers found 247 neurexin 1 isoforms: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3977267/

--Liz

**Comment by WayneForever** [comment]
The 288,773 isoforms are the results of collapse isoform. My input is the quivered fasta after Iso-Seq2 pipeline. The species of the project is Macaca.
Now I'm thinking that this result may be caused by high redundancy in the quivered fasta. And I'm try to reduce redundancy now. Thanks anyway. :)

**Comment by Magdoll** [comment]
Good to hear. Let me know what you find out.
